File: core/src/main/java/io/github/manami/core/services/events/AdvancedProgressState.java
package io.github.manami.core.services.events;

import io.github.manami.dto.entities.Anime;

/**
 * Contains the state as well as payload.
 */
public class AdvancedProgressState extends ProgressState {

  private Anime anime;


  public AdvancedProgressState(final int done, final int todo, final Anime anime) {
    super(done, todo);
    this.anime = anime;
  }

  public Anime getAnime() {
    return anime;
  }

  public void setAnime(Anime anime) {
    this.anime = anime;
  }
}

File: core/src/main/java/io/github/manami/core/tasks/events/AdvancedProgressState.kt
package io.github.manami.core.tasks.events

import io.github.manami.dto.entities.Anime

/**
 * Contains the state as well as payload.
 */
data class AdvancedProgressState(
        /**
         * Number of items which have already been processed.
         */
        val progressDone: Int,
        /**
         * Number of items which still need to be processed.
         */
        var progressTodo: Int,
        val anime: Anime
) : ProgressState(progressDone, progressTodo)
